tests/plugins/test_add_filesystem.py

```diff
@@ -496,7 +494,7 @@ def test_image_build_overwrites(tmpdir, architectures, architecture, reactor_con
         name = my-name
```

Comment:
I'd like to know which take precedence: 1) target from build param, or 2) target from image-build.conf. I think the way it's coded, target from image-build.conf will win. This is fine, it just means that after we get these changes in, we should work with internal RCM to clean up their image-build.conf; not a problem, safer actually.

Let's just add a test here to verify the behavior. Above, when `create_plugin_instance` is called, let's call it with a dummy target to make sure it's indeed ignored.

Reply:
Thanks for the review!
You are right. The target from `image-buiild.conf` overwrites the target from the build param. This way we will not break/change existing setups with the change and users would have to manually change their image-build.conf to use the feature.

About the test, on line 214 we now have:

`return AddFilesystemPlugin(tasker, workflow, KOJI_HUB, koji_target=KOJI_TARGET, **kwargs)`

So all calls to `create_plugin_instance` already set the target as defined by KOJI_TARGET and it is indeed changed from docker-f23 to docker-f23-candidate in that test. Should I still change the `create_plugin_instance` to explicitly receive the target?

Reply:
What happens if koji_target is not given? Will the target from image-build.conf get overridden with None?

Reply:
It is the opposite: the target value in the `image-build.conf`, when given, overwrites the koji target. If the koji_target is provided, it is always used.

If neither values are provided, we fail after parsing the config file, which was the behavior when no target was specified in `image-build.conf`.

If no koji_target is provided, but a target is specified in `image-build.conf`, the value in the file is used.

I added a few more tests for these cases.